### PR TITLE
Files in folder

### LINKED ISSUE #7: Simplify cli experience using directories
At this point we have a working program that takes 4 flags pointing to files and one of those files follows a specific format that users have to either copy/paste or just remember how to type out.

Let's change this up to work with directories. Let's get rid of the flags and instead use arguments to either create a new report folder or run the reports on a folder.

Remove the flags but keep the `flag.Parse()` and use `flag.Args` to check the arguments that the user passes in. If they run it with <2 or >2 arguments it should print some help text.

If they run it with the first argument being `new` and a second argument like

```ting-bill-split new 2017-02```

then make a folder named after the second argument and populate a template `bill.toml` file in it. Print a message telling the user to put `messages.csv`, `minutes.csv`, and `megabytes.csv` in the new folder.

If they run it with the first argument being `run` and a second argument like

```ting-bill-split run 2017-02```

Then it should look for the 4 files in that directory and do its processing.

### PR DESCRIPTION
**Pull request by hitjim**
* `ting-bill-split new <optional-name-param>` to create new dir, optionally providing a name
* `ting-bill-split dir <dir-name>` to run calculations based on files within provided directory.  Hint: works with `.` for current dir.

A little bit of duplicate code at the moment, but I'll reduce once it's all a little saner.  Maybe.  Probably?

Resolves #7

### CODE REVIEW

File: ting-bill-split.go [hunk 1]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
+	fmt.Printf("Ting Bill Splitter\n\n")
+
 	flag.Parse()
 	args := flag.Args()
 
 	if len(args) > 0 {
 		fmt.Println("Running in batch mode")
-		if args[0] == "new" {
+
+		command := args[0]
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Would be a good place to use `switch args[0]` instead of a bunch of if/elseif/else

File: ting-bill-split.go [hunk 2]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
 	}
 }
 
-func main() {
-	fmt.Printf("Ting Bill Splitter\n\n")
+func isFileMatch(fileName string, nameTerm string, ext string) bool {
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Can you add a comment describing how this function works?

File: ting-bill-split.go [hunk 3]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
+
+	if ext != "" {
+		r = regexp.MustCompile(`(?i)^[\w-]*` + nameTerm + `[\w-]*(\.` + ext + `)$`)
+	}
+
+	return r.MatchString(fileName)
+}
+
+func parseDir(path string) {
+	files, err := ioutil.ReadDir(path)
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Move this down after the `var` declarations so the error check is right after the function call

File: ting-bill-split.go [hunk 4]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
+	}
 
-	badParam := false
-	paramMap := map[string]*string{
-		"bills":     billPtr,
-		"minutes":   minPtr,
-		"messages":  msgPtr,
-		"megabytes": megPtr,
+	for _, file := range files {
+		if !file.IsDir() {
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Change this to 
```
if file.IsDir() {
    continue
}
```

and dedent the rest of this. de-indent? Un-indent? Undent? Back it up! lol

File: ting-bill-split.go [hunk 5]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
+		}
+
+		if msgFile == nil {
+			fmt.Println("Messages file not found.")
+		}
+
+		if megFile == nil {
+			fmt.Println("Megabytes file not found.")
+		}
+	} else {
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Don't use `else`. Make the previous block `return` if any of the files were missing.

File: ting-bill-split.go [hunk 6]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
 
-	badParam := false
-	paramMap := map[string]*string{
-		"bills":     billPtr,
-		"minutes":   minPtr,
-		"messages":  msgPtr,
-		"megabytes": megPtr,
+	for _, file := range files {
+		if !file.IsDir() {
+			fmt.Println(file.Name())
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Remove this

File: ting-bill-split.go [hunk 7]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
+
+			if billFile == nil && isFileMatch(file.Name(), "bills", "toml") {
+				billFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(billFile)
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Remove this

File: ting-bill-split.go [hunk 8]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
+			}
+
+			if minFile == nil && isFileMatch(file.Name(), "minutes", "csv") {
+				minFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(minFile)
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Remove this

File: ting-bill-split.go [hunk 9]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
+			}
+
+			if msgFile == nil && isFileMatch(file.Name(), "messages", "csv") {
+				msgFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(msgFile)
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Remove this

File: ting-bill-split.go [hunk 10]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
+			}
+
+			if megFile == nil && isFileMatch(file.Name(), "megabytes", "csv") {
+				megFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(megFile)
```
Comment (jcbwlkr) [inline ting-bill-split.go]: Remove this

File: ting-bill-split.go [hunk 11]
```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
...
-			fmt.Println("... or `-h` for flag options")
+			fmt.Println("Use `ting-bill-split new` or `new <billing-directory>` to create a new billing directory")
+			fmt.Println("Use `ting-bill-split dir <billing-directory>` to run on a directory containing a `bills.toml`, and CSV files for minutes, messages, and megabytes usage.")
+			fmt.Println("  Each of these files must contain their type somewhere in the filename - i.e. `YYYYMMDD-messages.csv` or `messages-potatosalad.csv` or whatever.")
+			fmt.Printf("\n... or `-h` for flag options")
 		}
 	} else {
 		fmt.Println("Running with with individual file assignments")
+
+		billPtr := flag.String("bills", "", "filename for bills toml - ex: -bills=\"bills.toml\"")
```
[the remark this answers is not in the document]
Comment (hitjim) [reply ting-bill-split.go]: don't worry @jcbwlkr I've got this

File: ting-bill-split.go (hunk 1 above)
Comment (hitjim) [reply ting-bill-split.go]: Why did I never think to look whether there was a `switch` for go?  Python brain, I guess.

File: ting-bill-split.go (hunk 3 above)
Comment (hitjim) [reply ting-bill-split.go]: You ever gotten highlighting like this?  Tests still pass, and now it stays that color anywhere I move it, even if I revert back to where it was per your comment above.

image

File: ting-bill-split.go (hunk 4 above)
Comment (hitjim) [reply ting-bill-split.go]: BACKED UP
*airhorn  and beeping sounds*

**Comment by hitjim** [comment]
😝💨 
@jcbwlkr

**Pull request merged**